tools/build_langserver/langserver/hover.go

```diff
@@ -29,29 +32,72 @@ func (h *LsHandler) handleHover(ctx context.Context, req *jsonrpc2.Request) (res
 	position := params.Position
 
 	h.mu.Lock()
-	content, err := getHoverContent(ctx, documentURI, position)
+	content, err := getHoverContent(ctx, h.analyzer, documentURI, position)
 	h.mu.Unlock()
 
 	if err != nil {
 		return nil, err
 	}
 
 	return &lsp.Hover{
-		Contents: content,
+		Contents: *content,
 	}, nil
 }
 
-func getHoverContent(ctx context.Context, uri lsp.DocumentURI, position lsp.Position) (content []lsp.MarkedString, err error) {
-	// Read file,
+func getHoverContent(ctx context.Context, analyzer *Analyzer, uri lsp.DocumentURI, position lsp.Position) (content *lsp.MarkupContent, err error) {
+	// Read file, and get a list of strings from the file
 	fileContent, err := ReadFile(ctx, uri)
 	if err != nil {
 		return nil, &jsonrpc2.Error{
 			Code: jsonrpc2.CodeParseError,
 			Message: fmt.Sprintf("fail to read file %s", uri),
 		}
 	}
-	// get the character from the line
-	fmt.Println(fileContent)
-	// look up the character from build_defs, and pull out the documentation
-	return nil, nil
+
+	// Get Hover Identifier
+	ident, err := analyzer.IdentFromPos(uri, position, fileContent)
+	if err != nil {
+		return nil, &jsonrpc2.Error{
+			Code: jsonrpc2.CodeParseError,
+			Message: fmt.Sprintf("fail to parse Build file %s", uri),
+		}
+	}
+
+	lineContent := fileContent[position.Line]
+
+	var contentString string
+	// check if the hovered line is an build target indentifier definition
+	if strings.Contains(lineContent, ident.Name) {
```

Comment:
why do you need to check this? doesn't `IdentFromPos` imply that this is always true (that the identifier you found will be in the line you think it is?

I think the thing you really do need to check here is that the identifier is in your set of builtins.

Reply:
This is more like checking if the hovered line is the start of the `Ident`. for example, if I hover over the line `go_library(`, then this condition would be true, in this case, I'd return the content for the definition, and the docstring.

However, if I hover over a line as so `deps = [`, then I'd be returning the content for the argument.

This is because the hovering content is per line based.